scripts/smithy.lic

```diff
@@ -0,0 +1,139 @@
+=begin
+
+  Smithy service helper script
+
+  This script will request multiple services from the smithy. It tries to be
+  strict about what it will accept for input and to verify that a blacksmith
+  NPC is present before requesting services.
+
+  Usage:
+
+  ;smithy <count> <crit, damage, or sighting>
+
+  author: Kragdruk
+  version: 1.0
+  tags: util, ccf, duskruin, festival, merchant
+
+=end
+
+module Smithy
+  SERVICE_TYPES = %w[critical damage sighting]
+
+  def self.determine_service_type_from(arg)
+    SERVICE_TYPES.find { |t| t.start_with? arg.downcase }
+  end
+
+  BLACKSMITH_DESCRIPTIONS = [
+    "old dwarven blacksmith",
+  ]
+
+  def self.blacksmith
+    GameObj.npcs.find do |npc|
+      BLACKSMITH_DESCRIPTIONS.include? npc.name
+    end
+  end
+
+  def self.print_usage
+    respond
+    respond "  Usage:"
+    respond
+    respond "  #{$clean_lich_char}#{Script.current.name} <count> <crit, damage, or sighting>"
+    respond
```

Comment:
How do you guys like to format your script's output? I use a lot of `echo` in my personal scripts because it is helpful for attribution when debugging, but it is kind of ugly for normal output during regular operation. How do you handle warnings/errors? My only real best practice here is to avoid `puts` because stdout is connected to the main lich process instead of the frontend when lich is running in detachable-client mode.

Reply:
I prefer `echo` over `respond` for same reason that it shows the script that sent the info.

Reply:
Heredocs make for much better Multiline output